## Research Questions:-

    Q1. Find the number of episodes?

    Q2. Max, min , mean of asked amount,asked equity, asked valuation? 

    Q3. Max & Min asked- equity,asked-valuation and asked amount episode-wise? 

    Q4. Brand names in which 2 ,3 or 4 sharks are invested? 

    Q5. Episode wise minimum and maximum of sharks invested? 

    Q6. Minimum and Maximum sharks invested in each brand? 

    Q7. Brand names who ask for 1 crore and got a deal? 

    Q8. Brand names where deal equity is between 0 to 50? 

    Q9. Find the number of brands participated in each episode? 

    Q10. How many sharks participated in this show and What were their names? 

    Q11. Find appearance of each sharks? 

    Q12. How many entrepreneurs were present? 

    Q13. How many times each shark invested the deal? 
    
    Q14. Find the equity percent that each sharks gets? 

    Q15. Find the total number of amount invested in this show? 

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv(r'C:\Users\HP\Desktop\Data Science\Data Sets For Practice\EDA_Shark_Tank_India.xlsx - Sheet1.csv')
df.head()

episode_number  pitch_number           brand_name  \
0               1             1  BluePine Industries   
1               1             2        Booz scooters   
2               1             3  Heart up my Sleeves   
3               2             4           Tagz Foods   
4               2             5       Head and Heart   

                                            idea  deal  pitcher_ask_amount  \
0                                   Frozen Momos     1                50.0   
1  Renting e-bike for mobility in private spaces     1                40.0   
2                             Detachable Sleeves     1                25.0   
3                           Healthy Potato Chips     1                70.0   
4                       Brain Development Course     0                50.0   

   ask_equity  ask_valuation  deal_amount  deal_equity  ...  ashneer_deal  \
0         5.0        1000.00         75.0        16.00  ...             1   
1        15.0         266.67         40.0        50.00  ...             1   
2        10.0         250.00         25.0        30.00  ...             0   
3         1.0        7000.00         70.0         2.75  ...             1   
4         5.0        1000.00          0.0         0.00  ...             0   

   anupam_deal  aman_deal  namita_deal  vineeta_deal  peyush_deal  \
0            0          1            0             1            0   
1            0          0            0             1            0   
2            1          0            0             1            0   
3            0          0            0             0            0   
4            0          0            0             0            0   

   ghazal_deal  total_sharks_invested  amount_per_shark  equity_per_shark  
0            0                      3              25.0          5.333333  
1            0                      2              20.0         25.000000  
2            0                      2              12.5         15.000000  
3            0                      1              70.0          2.750000  
4            0                      0               0.0          0.000000  

[5 rows x 28 columns]

In [3]:
df.shape

(117, 28)

#  Q1. Find the number of episodes?

In [4]:
df['episode_number'].max()

35

# Q2. Max, min , mean of asked amount,asked equity, asked valuation? 

In [5]:
df.columns

Index(['episode_number', 'pitch_number', 'brand_name', 'idea', 'deal',
       'pitcher_ask_amount', 'ask_equity', 'ask_valuation', 'deal_amount',
       'deal_equity', 'deal_valuation', 'ashneer_present', 'anupam_present',
       'aman_present', 'namita_present', 'vineeta_present', 'peyush_present',
       'ghazal_present', 'ashneer_deal', 'anupam_deal', 'aman_deal',
       'namita_deal', 'vineeta_deal', 'peyush_deal', 'ghazal_deal',
       'total_sharks_invested', 'amount_per_shark', 'equity_per_shark'],
      dtype='object')

In [6]:
min_value  = df['pitcher_ask_amount'].min()
max_value  = df['pitcher_ask_amount'].max()
mean_value  = df['pitcher_ask_amount'].mean()


print("Min Amount Asked by Pitcher:", min_value)
print("Max Amount Asked by Pitcher:", max_value)
print("Mean Amount Asked by Pitcher:", mean_value)

Min Amount Asked by Pitcher: 0.00101
Max Amount Asked by Pitcher: 30000.0
Mean Amount Asked by Pitcher: 319.8547094871795


In [7]:
min_value  = df['ask_equity'].min()
max_value  = df['ask_equity'].max()
mean_value  = df['ask_equity'].mean()


print("Min Equity Asked:", min_value)
print("Max Equity Asked:", max_value)
print("Mean Equity Asked:", mean_value)

Min Equity Asked: 0.25
Max Equity Asked: 25.0
Mean Equity Asked: 5.188034188034188


In [8]:
min_value  = df['ask_valuation'].min()
max_value  = df['ask_valuation'].max()
mean_value  = df['ask_valuation'].mean()


print("Min Valuation Asked:", min_value)
print("Max Valuation Asked:", max_value)
print("Mean Valuation Asked:", mean_value)

Min Valuation Asked: 0.01
Max Valuation Asked: 120000.0
Mean Valuation Asked: 3852.4624786324785


# Q3. Max & Min asked- equity,asked-valuation and asked amount episode-wise? 

In [9]:
min_value = df.loc[df['ask_equity']==df['ask_equity'].min(), 'episode_number']
max_value = df.loc[df['ask_equity']==df['ask_equity'].max(), 'episode_number']

print("Min Equity Asked in Episode:", min_value)
print("Max Equity Asked in Episode:", max_value)

Min Equity Asked in Episode: 6    3
Name: episode_number, dtype: int64
Max Equity Asked in Episode: 30    11
Name: episode_number, dtype: int64


* In row no. 6 and Episode no. 3 minimum equity was asked.
* In row no. 30 and Episode no. 11 maximum equity was asked.

In [10]:
min_value = df.loc[df['ask_valuation']==df['ask_valuation'].min(), 'episode_number']
max_value = df.loc[df['ask_valuation']==df['ask_valuation'].max(), 'episode_number']

print("Min Equity Asked in Episode:", min_value)
print("Max Equity Asked in Episode:", max_value)

Min Equity Asked in Episode: 85    27
Name: episode_number, dtype: int64
Max Equity Asked in Episode: 30    11
Name: episode_number, dtype: int64


* In row no. 85 and Episode no. 27 minimum Valuation was asked.
* In row no. 30 and Episode no. 11 maximum Valuation was asked.

In [11]:
min_value = df.loc[df['pitcher_ask_amount']==df['pitcher_ask_amount'].min(), 'episode_number']
max_value = df.loc[df['pitcher_ask_amount']==df['pitcher_ask_amount'].max(), 'episode_number']

print("Min Equity Asked in Episode:", min_value)
print("Max Equity Asked in Episode:", max_value)

Min Equity Asked in Episode: 85    27
Name: episode_number, dtype: int64
Max Equity Asked in Episode: 30    11
Name: episode_number, dtype: int64


* In row no. 85 and Episode no. 27 minimum Amount was asked.
* In row no. 30 and Episode no. 11 maximum Amount was asked.

## Conclusion:-


* Min Amount, Valuation asked by Pitcher in Episode No. 27


* Max Amount, Valuation, Equity asked by Pitcher in Episode No. 11

# Q4.Brand names in which 2 ,3 or 4 sharks are invested?

In [12]:
x = df.loc[df['total_sharks_invested']==2, 'brand_name'].tolist()
x

['Booz scooters',
 'Heart up my Sleeves',
 'Cosiq',
 'Bummer',
 'Revamp Moto',
 'Raising Superstars',
 'Beyond Snack',
 'Vivalyf Innovations- Easy Life',
 'Altor',
 'Ariro',
 'Farda',
 'Caragreen',
 'The Renal Project',
 'Bamboo India',
 'Beyond Water',
 "Let's Try",
 'The Quirky Nari',
 'The Sass Bar',
 'Kabaddi Adda',
 'Proxgy']

* Only 2 Sharks Invested in above Brands

In [13]:
x = df.loc[df['total_sharks_invested']==3, 'brand_name'].tolist()
x

['BluePine Industries',
 'Nuutjob',
 'Meatyour',
 'EventBeep',
 'LOKA',
 'Annie',
 'Cocofit',
 'Aas Vidyalaya',
 'Get a Whey',
 'Hair Originals',
 'Humpy A2',
 'Gold Safe Solutions Ind.',
 'Wakao Foods',
 'Tweek Labs']

* 3 Sharks Invested in above brands

In [14]:
x = df.loc[df['total_sharks_invested']==4, 'brand_name'].tolist()
x

['The Yarn Bazaar',
 'WeSTOCK',
 'Watt Technovations',
 'Nomad Food Project',
 'Jain Shikanji']

* 4 Sharks Invested in above brands

# Q5. Episode in which minimum and maximum of sharks invested? 

In [15]:
df.total_sharks_invested.value_counts()

0    52
1    22
2    20
3    14
4     5
5     4
Name: total_sharks_invested, dtype: int64

* As we see minimum no. of sharks invested is 1 and maximum sharks invested are 5

In [16]:
x = df.loc[df['total_sharks_invested']==1, 'episode_number'].tolist()
print(x)

[2, 3, 3, 6, 8, 11, 12, 15, 15, 17, 18, 20, 21, 23, 24, 25, 26, 26, 27, 31, 32, 33]


Only 1 shark was invested in above episodes.

In [17]:
x = df.loc[df['total_sharks_invested']==5, 'episode_number'].tolist()
print(x)

[6, 17, 20, 25]


5 shark was invested in above episodes.

# Q6. Minimum and Maximum sharks invested in each brand?

In [18]:
x = df.loc[df['total_sharks_invested']==1, 'brand_name'].tolist()
x

['Tagz Foods',
 'Peeschute',
 'NOCD',
 'Menstrupedia',
 'Motion Breeze',
 'ARRCOAT Surface Textures',
 'Auli Lifestyle',
 'Hammer Lifestyle',
 'PNT',
 'RoadBounce',
 'Otua',
 'The State Plate',
 'Sid07 Designs',
 'Namhya Foods',
 'KG Agrotech',
 'PawsIndia',
 'Isak Fragrances',
 'Rare Planet',
 'Insurance Samadhan',
 'AyuRythm',
 'GrowFitter',
 'Colour Me Mad']

* Only 1 Shark was invested in above brands.

In [19]:
x = df.loc[df['total_sharks_invested']==5, 'brand_name'].tolist()
x

['Skippi Pops', 'Find Your Kicks India', 'IN A CAN', 'Sunfox Technologies']

* 5 Sharks was invested in above Brands

# Q7. Brand names who ask for 1 crore and got a deal?

In [57]:
df2 = df.loc[(df['deal'] == 1) & (df['pitcher_ask_amount'] == 100.0) & (df['brand_name'])]
df2 = df2['brand_name'].tolist()
df2

['Revamp Moto',
 'The Renal Project',
 'Otua',
 'Get a Whey',
 'Namhya Foods',
 'Sunfox Technologies',
 'Insurance Samadhan']

### OR

In [55]:
# Filter the data for brand names who asked for 1 crore
df = df[df["pitcher_ask_amount"] == 100.0]

# Filter the data for brand names who got a deal
df = df[df["deal"] == 1]

# Get the brand names
brand_names = df["brand_name"].unique().tolist()
brand_names

['Revamp Moto',
 'The Renal Project',
 'Otua',
 'Get a Whey',
 'Namhya Foods',
 'Sunfox Technologies',
 'Insurance Samadhan']

* These are brands who ask for 1 Crore and Got a deal.

#  Q8. Brand names where deal equity is between 0 to 50? 

In [22]:
df2 = df.loc[(df['deal'] == 1) & (df['equity_per_shark'] < 50) & (df['brand_name'])]
df2 = pd.DataFrame(df2)
df2 = df2[['brand_name','deal_equity']]
df2

brand_name  deal_equity
0    BluePine Industries        16.00
1          Booz scooters        50.00
2    Heart up my Sleeves        30.00
3             Tagz Foods         2.75
7              Peeschute         6.00
..                   ...          ...
106        Colour Me Mad        25.00
108           Tweek Labs        10.00
109               Proxgy        10.00
110   Nomad Food Project        20.00
114        Jain Shikanji        30.00

[63 rows x 2 columns]

In [23]:
df2['brand_name'].values

array(['BluePine Industries', 'Booz scooters', 'Heart up my Sleeves',
       'Tagz Foods', 'Peeschute', 'NOCD', 'Cosiq', 'Bummer',
       'Revamp Moto', 'Skippi Pops', 'Menstrupedia', 'Torch-it',
       'Beyond Snack', 'Vivalyf Innovations- Easy Life', 'Motion Breeze',
       'Altor', 'Ariro', 'Nuutjob', 'Meatyour', 'EventBeep',
       'ARRCOAT Surface Textures', 'Farda', 'Auli Lifestyle', 'LOKA',
       'Annie', 'Caragreen', 'The Yarn Bazaar', 'The Renal Project',
       'Hammer Lifestyle', 'PNT', 'Cocofit', 'Bamboo India',
       'Beyond Water', "Let's Try", 'Find Your Kicks India',
       'Aas Vidyalaya', 'RoadBounce', 'Otua', 'WeSTOCK',
       'The State Plate', 'IN A CAN', 'Get a Whey', 'The Quirky Nari',
       'Hair Originals', 'Namhya Foods', 'The Sass Bar', 'KG Agrotech',
       'PawsIndia', 'Sunfox Technologies', 'Rare Planet',
       'Watt Technovations', 'Insurance Samadhan', 'Humpy A2',
       'Gold Safe Solutions Ind.', 'Wakao Foods', 'Kabaddi Adda',
       'AyuRythm', 'G

In [24]:
len(df2['brand_name'])

63

* There are 63 brand which gets a deal and their deal equity is between 0-50%. Their names are given above.

# Q9. Find the number of brands participated in each episode?

In [25]:
brands_per_episode = df.groupby('episode_number')['brand_name'].nunique()

In [26]:
df2 = pd.DataFrame(brands_per_episode)
df2

brand_name
episode_number            
1                        3
2                        3
3                        3
4                        3
5                        3
6                        3
7                        3
8                        3
9                        3
10                       3
11                       3
12                       3
13                       3
14                       3
15                       3
16                       4
17                       4
18                       4
19                       4
20                       3
21                       3
22                       4
23                       4
24                       3
25                       3
26                       3
27                       4
28                       3
29                       3
30                       4
31                       4
32                       4
33                       4
34                       4
35                       3

#  Q10. How many sharks participated in this show and What were their names? 

In [27]:
age_cols = [col for col in df.columns if 'present' in col]
df3 = pd.DataFrame(age_cols)
df3

0
0  ashneer_present
1   anupam_present
2     aman_present
3   namita_present
4  vineeta_present
5   peyush_present
6   ghazal_present

These are Shark's name who participated in the show.

In [28]:
df3.nunique()

0    7
dtype: int64

There are total 7 Sharks 

# Q11. Find appearance of each sharks?

## 1) Ghazal appear in these Episodes

In [29]:
abc = df.loc[df['ghazal_present']==1, 'episode_number']

df3 = pd.DataFrame(abc)

df3 = df3.drop_duplicates(subset=['episode_number'])

df3

episode_number
75               24
78               25
81               26
84               27
88               28
91               29
110              34
114              35

In [30]:
len(df3['episode_number'])

8

Ghazal Appear in 8 Episodes out of 35 Episodes which are given above.

## 2) Ashneer appear in these Episodes

In [31]:
abc = df.loc[df['ashneer_present']==1, 'episode_number']

df4 = pd.DataFrame(abc)

df4 = df4.drop_duplicates(subset=['episode_number'])

df4

episode_number
0                 1
3                 2
6                 3
9                 4
12                5
15                6
18                7
21                8
24                9
27               10
30               11
33               12
36               13
39               14
42               15
45               16
49               17
53               18
57               19
61               20
64               21
67               22
71               23
94               30
98               31
102              32
106              33
110              34
114              35

In [32]:
len(df4['episode_number'])

29

Asneer Appear in 29 Episodes out of 35 Episodes which are given above.

 ## 3) Anupam Appears in these Episodes

In [33]:
abc = df.loc[df['anupam_present']==1, 'episode_number']

df5 = pd.DataFrame(abc)

df5 = df5.drop_duplicates(subset=['episode_number'])

df5

episode_number
0                 1
3                 2
6                 3
9                 4
12                5
15                6
18                7
21                8
24                9
27               10
30               11
33               12
36               13
39               14
42               15
45               16
49               17
53               18
57               19
61               20
64               21
67               22
71               23
75               24
78               25
81               26
84               27
88               28
91               29
94               30
98               31
102              32
106              33
110              34
114              35

In [34]:
len(df5['episode_number'])

35

Anupam Appear in 35 Episodes out of 35 Episodes which are given above.

## 4) Aman appears in these Episodes

In [35]:
abc = df.loc[df['aman_present']==1, 'episode_number']

df5 = pd.DataFrame(abc)

df5 = df5.drop_duplicates(subset=['episode_number'])

df5

episode_number
0                 1
3                 2
6                 3
9                 4
12                5
15                6
18                7
21                8
24                9
27               10
30               11
33               12
36               13
39               14
42               15
45               16
49               17
53               18
57               19
61               20
64               21
67               22
71               23
94               30
98               31
102              32
106              33
110              34
114              35

In [36]:
len(df5['episode_number'])

29

Aman Appear in 29 Episodes out of 35 Episodes which are given above.

## 5) Namita appears in these Episodes

In [37]:
abc = df.loc[df['namita_present']==1, 'episode_number']

df6 = pd.DataFrame(abc)

df6 = df6.drop_duplicates(subset=['episode_number'])

df6

episode_number
0                 1
3                 2
6                 3
9                 4
12                5
15                6
18                7
21                8
24                9
27               10
30               11
33               12
36               13
39               14
42               15
45               16
49               17
53               18
57               19
61               20
75               24
78               25
81               26
84               27
88               28
91               29
94               30
98               31
102              32
106              33
110              34
114              35

In [38]:
len(df6['episode_number'])

32

Namita Appear in 32 Episodes out of 35 Episodes which are given above.

## 6) Peyush appears in these Episodes

In [39]:
abc = df.loc[df['peyush_present']==1, 'episode_number']

df7 = pd.DataFrame(abc)

df7 = df7.drop_duplicates(subset=['episode_number'])

df7

episode_number
21                8
24                9
27               10
30               11
33               12
36               13
39               14
42               15
45               16
49               17
53               18
57               19
61               20
64               21
67               22
71               23
75               24
78               25
81               26
84               27
88               28
91               29
102              32
106              33
110              34
114              35

In [40]:
len(df7['episode_number'])

26

Peyush Appear in 26 Episodes out of 35 Episodes which are given above.

## 7) Vineeta Appears in These Episodes

In [41]:
abc = df.loc[df['vineeta_present']==1, 'episode_number']

df8 = pd.DataFrame(abc)

df8 = df8.drop_duplicates(subset=['episode_number'])

df8

episode_number
0                 1
3                 2
6                 3
9                 4
12                5
15                6
18                7
64               21
67               22
71               23
75               24
78               25
81               26
84               27
88               28
91               29
94               30
98               31
110              34
114              35

In [42]:
len(df8['episode_number'])

20

Vineeta Appear in 20 Episodes out of 35 Episodes which are given above.

# Q12. How many entrepreneurs were present? 

In [43]:
df.pitch_number.max()

117

* There are total 117 entrepreneurs presented who pitched their business.

# Q13. How many times each shark invested the deal?

In [44]:
df.ashneer_deal.value_counts()

0    96
1    21
Name: ashneer_deal, dtype: int64

In [45]:
df.anupam_deal.value_counts()

0    93
1    24
Name: anupam_deal, dtype: int64

In [46]:
df.aman_deal.value_counts()

0    89
1    28
Name: aman_deal, dtype: int64

In [47]:
df.namita_deal.value_counts()

0    95
1    22
Name: namita_deal, dtype: int64

In [48]:
df.vineeta_deal.value_counts()

0    102
1     15
Name: vineeta_deal, dtype: int64

In [49]:
df.peyush_deal.value_counts()

0    90
1    27
Name: peyush_deal, dtype: int64

In [50]:
df.ghazal_deal.value_counts()

0    110
1      7
Name: ghazal_deal, dtype: int64

* Ashneer invested 21 times.


* Anupam invested 24 times.


* Aman invested 28 times.


* Namita invested 22 times.


* Vineeta invested 15 times.


* Peyush invested 27 times.


* Ghazal invested 7 times.

# Q14. Find the equity percent that each sharks gets? 

In [51]:
df10 = df.loc[(df['deal'] == 1) & (df['equity_per_shark'] > 0) & (df['brand_name'])]
df10 = pd.DataFrame(df10)
X = df10[['brand_name','equity_per_shark']]
X

brand_name  equity_per_shark
0    BluePine Industries          5.333333
1          Booz scooters         25.000000
2    Heart up my Sleeves         15.000000
3             Tagz Foods          2.750000
7              Peeschute          6.000000
..                   ...               ...
106        Colour Me Mad         25.000000
108           Tweek Labs          3.333333
109               Proxgy          5.000000
110   Nomad Food Project          5.000000
114        Jain Shikanji          7.500000

[64 rows x 2 columns]

These are the brands which gets the deal and Equity Percentage that each shark get

# Q15. Find the total number of amount invested in this show? 

In [52]:
df.deal_amount.sum()

3742.00106

Total Amount Invested by all Sharks in the Show is 37 Crores 42 Lakh 106 Rupees 